sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2505,27 +2508,27 @@ def _stage_block_options(
 
         options = {
             'block_id': block_id,
-            'content_length': length,
+            'content_length': content_length,
             'body': data,
-            'transactional_content_md5': None,
             'timeout': kwargs.pop('timeout', None),
             'lease_access_conditions': access_conditions,
-            'validate_content': validate_content,
+            'validate_content': True if validate_content is True or validate_content == ChecksumAlgorithm.MD5 else None,
```

Comment:
Maintaining backwards compatibility sort of makes this feel odd.
validate_content=True --> MD5
validate_content=ChecksumAlgorithm.MD5 --> MD5
validate_content=False --> off
validate_content=None --> As of now, CRC64 but could be any others in the future
Note: These are regarding what gets set in `options['validate_content']` not the actual `validate_content` kwarg itself

It sort of feels odd that we pass in `None` for `validate_content` even though we technically _are_ validating content, but we can't put `True` because that meant something else previously 😢

Reply:
Yeah, it's a bit unfortunate but just how it has to be for legacy reasons I think. Previously content validation was handled through a pipeline policy which looked for `validate_content` in the request context and the way we add something to the request context as by passing it as a `kwarg` (any unrecognized thing will get added to the request context I think). `None` essentially tells the pipeline policy to do nothing because we are handling it ourselves.